### PR TITLE
chore: drop Python 3.8

### PR DESCRIPTION
**Pull request by SdgJlbl**

### CODE REVIEW

File: .github/workflows/publish_docker.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     outputs:
       # Add a version to this list to automatically build Dockerfiles from it for the next releases
       default_cuda_images: '["nvidia/cuda:11.8.0-base-ubuntu22.04"]'
```
Comment (thbcmlowk) [inline .github/workflows/publish_docker.yml:31]: Maybe not directly related to dropping py3.8 but rather supporting py3.11 and the [python-slim switch](https://github.com/Substra/substrafl/pull/197) but should we not rework this workflow in consequence?

Comment (thbcmlowk) [inline .github/workflows/publish_docker.yml:31]: Maybe not directly related to dropping py3.8 but rather supporting py3.11 and the [python-slim switch](https://github.com/Substra/substrafl/pull/197) but should we not rework this workflow in consequence?

```suggestion
```

Comment (SdgJlbl) [reply .github/workflows/publish_docker.yml:31]: If we are no longer publishing custom substra-tools images, I think we should drop this action entirely 🤔 

Do we want to do that in this PR, or still discuss it further?

Comment (ThibaultFy) [reply .github/workflows/publish_docker.yml:31]: Agree with droping this action if don't use substra-tools base image anywhere. But definitely not in this pr :/

**Review by ThibaultFy — COMMENTED** [review]
LGTM, but one small comment about `pypa/gh-action-pypi-publish@v1.5.1`. Might not be compatible with 3.11 as they pass to 3.11 in [1.6.0](https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.6.0). Might worth to upgrade to latest ([v1.8.14](https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.8.14)).

Or we can let it like this and test on next release.

**Comment by SdgJlbl** [comment]
> LGTM, but one small comment about `pypa/gh-action-pypi-publish@v1.5.1`. Might not be compatible with 3.11 as they pass to 3.11 in [1.6.0](https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.6.0). Might worth to upgrade to latest ([v1.8.14](https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.8.14)).
> 
> Or we can let it like this and test on next release.

Good catch, I've update GHA versions, thanks 🙏

**Review by ThibaultFy — APPROVED** [review]
LGTM :)

File: .github/workflows/publish_docker.yml (hunk 1 above)
Comment (SdgJlbl) [reply .github/workflows/publish_docker.yml:31]: Created a follow-up card https://linear.app/owkin/issue/FL-1511/do-we-want-to-drop-substra-tools-base-image

**Pull request merged**